src/oversampling_methods/borderline_smote1/interface_mlj.jl

```diff
@@ -0,0 +1,169 @@
+
+### BorderlineSMOTE1 with MLJ Interface
+
+mutable struct BorderlineSMOTE1{T,R<:Union{Integer,AbstractRNG}, I<:Integer} <: Static
+    m::I
+    k::I
+    ratios::T
+    rng::R
+    try_perserve_type::Bool
+end;
+
+
+
+"""
+Check whether the given model hyperparameters are valid and clean them if necessary. 
+"""
+function MMI.clean!(s::BorderlineSMOTE1)
+    message = ""
+    if s.k < 1
+        throw(ERR_NONPOS_K(s.k))
+    end
+    if s.m < 1
+        throw(ERR_NONPOS_K(s.m))
+    end
+    return message
+end
+
+
+
+
+"""
+Initiate a BorderlineSMOTE1 model with the given hyper-parameters.
+"""
+function BorderlineSMOTE1(;
+    m::Integer = 5,
+    k::Integer = 5,
+    ratios::Union{Nothing,AbstractFloat,Dict{T,<:AbstractFloat}} = 1.0,
+    rng::Union{Integer,AbstractRNG} = default_rng(), try_perserve_type::Bool=true
+) where {T}
```

Comment:
Not wrong, but you can also do

```suggestion
    ratios::Union{Nothing,AbstractFloat,Dict{<:Any,<:AbstractFloat}} = 1.0,
    rng::Union{Integer,AbstractRNG} = default_rng(), try_perserve_type::Bool=true
)
```

Reply:
Thank you. I am aware of this.

Reply:
@ablaom I think I have been copying this line over in all methods. So, I may be more concise by writing it in the second way for them all after merging.

Reply:
Really no need to change. This was just a FYI